library(methylKit)
bam.list = list("/home/huchenyuan/mouse_project/methylation_analysis/not_duplicate/SRR833527_CpG.txt","/home/huchenyuan/mouse_project/methylation_analysis/not_duplicate/SRR833528_CpG.txt")
myjob = methRead(bam.list, sample.id = list("SRR833527","SRR833528"), assembly="Mouse", treatment=c(0,1), context="CpG")
getMethylationStats(myjob[[1]],plot=TRUE,both.strands=FALSE)
getMethylationStats(myjob[[2]],plot=TRUE,both.strands=FALSE)
getCoverageStats(myjob[[1]],plot = TRUE,both.strands = FALSE)
getCoverageStats(myjob[[2]],plot = TRUE,both.strands = FALSE)
meth = unite(myjob,destrand = FALSE)
getCorrelation(meth,plot = TRUE)
clusterSamples(meth,dist = "correlation",method = "ward",plot = TRUE)
PCASamples(meth,screeplot= TRUE)
PCASamples(meth)
tiles=tileMethylCounts(myjob,win.size=500,step.size=500)
meth1=unite(tiles, destrand=FALSE)
myDiff=calculateDiffMeth(meth1,num.cores=12)
myDiff25p.hyper=getMethylDiff(myDiff,difference=25,qvalue=0.01,type="hyper")
myDiff25p.hypo=getMethylDiff(myDiff,difference=25,qvalue=0.01,type="hypo")
myDiff25p=getMethylDiff(myDiff,difference=25,qvalue=0.01)
diffMethPerChr(myDiff,plot=FALSE,qvalue.cutoff=0.01, meth.cutoff=25)
diffMethPerChr(myDiff,plot=TRUE,qvalue.cutoff=0.01, meth.cutoff=25)
bam.list = list("/home/huchenyuan/mouse_project/methylation_analysis/not_duplicate/SRR833527_CpG_filter.txt","/home/huchenyuan/mouse_project/methylation_analysis/not_duplicate/SRR833528_CpG_filter.txt")
myjob = methRead(bam.list, sample.id = list("SRR833527","SRR833528"), assembly="Mouse", treatment=c(0,1), context="CpG")
tiles=tileMethylCounts(myjob,win.size=500,step.size=500)
meth1=unite(tiles, destrand=FALSE)
myDiff=calculateDiffMeth(meth1,num.cores=12)
myDiff25p.hyper=getMethylDiff(myDiff,difference=25,qvalue=0.01,type="hyper")
myDiff25p.hypo=getMethylDiff(myDiff,difference=25,qvalue=0.01,type="hypo")
myDiff25p=getMethylDiff(myDiff,difference=25,qvalue=0.01)
diffMethPerChr(myDiff,plot=TRUE,qvalue.cutoff=0.01, meth.cutoff=25)
diffMethPerChr(myDiff,plot=TRUE,qvalue.cutoff=0.01, meth.cutoff=25)
diffMethPerChr(myDiff,plot=FALSE,qvalue.cutoff=0.01, meth.cutoff=25)
diffMethPerChr(myDiff,plot=TRUE,qvalue.cutoff=0.01, meth.cutoff=25)
save.image("hyper_hypo")
library(genomation)
BiocManager::install("genomation")
library(genomation)
gene.obj=readTranscriptFeatures("/home/huchenyuan/mouse_project/methylation_analysis/reference.txt")
diffAnn=annotateWithGeneParts(as(myDiff25p,"GRanges"),gene.obj)
bam.list = list("/home/huchenyuan/mouse_project/methylation_analysis/not_duplicate/SRR833527_CpG_filter.txt","/home/huchenyuan/mouse_project/methylation_analysis/not_duplicate/SRR833528_CpG_filter.txt")
myjob = methRead(bam.list, sample.id = list("SRR833527","SRR833528"), assembly="Mouse", treatment=c(0,1), context="CpG")
tiles=tileMethylCounts(myjob,win.size=500,step.size=500)
meth1=unite(tiles, destrand=FALSE)
myDiff=calculateDiffMeth(meth1,num.cores=12)
myDiff=calculateDiffMeth(meth1)
tiles=tileMethylCounts(myjob,win.size=1000,step.size=1000)
meth1=unite(tiles, destrand=FALSE)
myDiff=calculateDiffMeth(meth1)
myDiff25p=getMethylDiff(myDiff,difference=25,qvalue=0.01)
gene.obj=readTranscriptFeatures("/home/huchenyuan/mouse_project/methylation_analysis/reference.txt")
diffAnn=annotateWithGeneParts(as(myDiff25p,"GRanges"),gene.obj)
diffann = annotateDMRInfo(as(myDiff25p,"GRanges"),'TxDb.Mmusculus.UCSC.mm10.knownGene')
library(methyAnalysis)
BiocManager::install("methyAnalysis")
gene.obj = readTranscriptFeatures("/home/huchenyuan/mouse_project/methylation_analysis/reference.txt")
diffAnn = annotateWithGeneParts(as(myDiff25p,"GRanges"),gene.obj)
gene.obj = readTranscriptFeatures("/home/huchenyuan/mouse_project/methylation_analysis/reference.txt")
diffAnn = annotateWithGeneParts(as(myDiff25p,"GRanges"),gene.obj)
gene.obj = readTranscriptFeatures("/home/huchenyuan/mouse_project/methylation_analysis/reference.txt")
diffAnn = annotateWithGeneParts(as(myDiff25p,"GRanges"),gene.obj)
write.csv(getAssociationWithTSS(diffAnn),"P11m-P3m.csv")
getTargetAnnotationStats(diffAnn,percentage=TRUE,precedence=TRUE)
plotTargetAnnotation(diffAnn,precedence=TRUE, main="differential methylation annotation")
getMethylationStats(myjob[[1]],plot=TRUE,both.strands=FALSE)
getMethylationStats(myjob[[2]],plot=TRUE,both.strands=FALSE)
getMethylationStats(myjob[[1]],plot=TRUE,both.strands=FALSE)
getCoverageStats(myjob[[1]],plot = TRUE,both.strands = FALSE)
getCoverageStats(myjob[[2]],plot = TRUE,both.strands = FALSE)
meth = unite(myjob,destrand = FALSE)
PCASamples(meth,screeplot= TRUE)
PCASamples(meth)
getCorrelation(meth,plot = TRUE)
clusterSamples(meth,dist = "correlation",method = "ward",plot = TRUE)
myDiff25p.hyper=getMethylDiff(myDiff,difference=25,qvalue=0.01,type="hyper")
myDiff25p.hypo=getMethylDiff(myDiff,difference=25,qvalue=0.01,type="hypo")
diffMethPerChr(myDiff,plot=TRUE,qvalue.cutoff=0.01, meth.cutoff=25)
q()
